components/vfs/manager.cpp

```diff
@@ -107,4 +108,31 @@ namespace VFS
         normalize_path(name, mStrict);
     }
 
+    const std::string& Manager::findFirstOf(const std::string &name) const
+    {
+        std::string normalized = name;
```

Comment:
If you find yourself doing this (taking a const reference parameter which is copied to a local and the original never touched again), it's better to take the parameter by value:
```
const std::string& Manager::findFirstOf(std::string normalized) const
{
    normalize_path(normalized, mStrict);
    ...
```
This has the benefit of potentially avoiding an extra copy if a temporary is bound to the parameter reference. e.g.
```
mgr->findFirstOf(pathstr + namestr);
```
As it is, the `operator+` returns a new string value, which is moved to a temporary and bound to a const reference, which is then copied to a local variable to use. However, with the parameter taken by value, the value returned by `operator+` is moved to the parameter value (C++ actually says even this move can be elided, constructing the return value in place of the parameter), which the method then uses directly. Only the necessary amount of copies are made, and they can be handled more efficiently since the compiler is aware of their lifetimes.

Reply:
Oh, cool. Thanks.